### PR TITLE
Add GitHub Actions

### PR DESCRIPTION
**Pull request by y-yu**
~https://github.com/y-yu/grpc-ocaml/runs/6654576467?check_suite_focus=true~

<details>
```
Run opam exec -- dune build
(cd _build/default && /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/bin/ocamlc.opt -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I server/.grpc_server.objs/byte -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/angstrom -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/base_internalhash_types -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/caml -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/md5 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/shadow_stdlib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base64 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base_bigstring -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base_quickcheck -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bigarray-compat -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bigstringaf -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bin_prot -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bin_prot/shape -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/core -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/core/base_for_tests -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/core/validate -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/faraday -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/faraday-lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/faraday-lwt-unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/fieldslib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/gluten -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/gluten-lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/gluten-lwt-unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/h2 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/h2-lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/h2-lwt-unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/hmap -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/hpack -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/httpaf -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/int_repr -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/jane-street-headers -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/logs -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/lwt/unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/lwt_ssl -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/mmap -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocaml-protoc-plugin -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocaml/threads -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocplib-endian -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocplib-endian/bigstring -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/parsexp -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_assert/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_bench/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_compare/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_deriving/runtime -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_enumerate/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/collector -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/common -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/config -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/config_types -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_hash/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_here/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_inline_test/config -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_inline_test/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_log/types -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_module_timer/runtime -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_sexp_conv/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/psq -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/result -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/seq -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/sexplib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/sexplib0 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/splittable_random -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ssl -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/stdio -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/stringext -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/time_now -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/typerep -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/uri -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/variantslib -I basic/.Grpc_basic.objs/byte -no-alias-deps -opaque -open Grpc_server__ -o server/.grpc_server.objs/byte/grpc_server__Handler.cmo -c -impl server/handler.pp.ml)
File "server/handler.ml", line 23, characters 8-83:
23 |         (module PB.Service.Rpc with type Request.t = req and type Response.t = res)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Unbound module type PB.Service.Rpc
(cd _build/default && /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/bin/ocamlc.opt -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I client/.grpc_client.objs/byte -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/angstrom -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/base_internalhash_types -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/caml -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/md5 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base/shadow_stdlib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base64 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base_bigstring -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base_quickcheck -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bigarray-compat -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bigstringaf -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bin_prot -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/bin_prot/shape -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/core -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/core/base_for_tests -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/core/validate -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/faraday -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/faraday-lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/faraday-lwt-unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/fieldslib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/gluten -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/gluten-lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/gluten-lwt-unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/h2 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/h2-lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/h2-lwt-unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/hpack -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/httpaf -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/int_repr -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/jane-street-headers -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/logs -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/lwt -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/lwt/unix -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/lwt_ssl -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/mmap -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocaml-protoc-plugin -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocaml/threads -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocplib-endian -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ocplib-endian/bigstring -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/parsexp -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_assert/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_bench/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_compare/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_deriving/runtime -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_enumerate/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/collector -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/common -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/config -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_expect/config_types -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_hash/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_here/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_inline_test/config -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_inline_test/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_log/types -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_module_timer/runtime -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ppx_sexp_conv/runtime-lib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/psq -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/result -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/seq -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/sexplib -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/sexplib0 -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/splittable_random -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/ssl -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/stdio -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/stringext -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/time_now -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/typerep -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/uri -I /home/runner/work/grpc-ocaml/grpc-ocaml/_opam/lib/variantslib -I basic/.Grpc_basic.objs/byte -no-alias-deps -opaque -open Grpc_client__ -o client/.grpc_client.objs/byte/grpc_client__Client.cmo -c -impl client/client.pp.ml)
File "client/client.ml", lines 119-121, characters 6-37:
119 | ......(module Ocaml_protoc_plugin.Service.Rpc
120 |          with type Request.t = 'req
121 |           and type Response.t = 'res)
Error: Unbound module type Ocaml_protoc_plugin.Service.Rpc
```
</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        uses: actions/checkout@v2
+
+      - name: Setup OCaml ${{ matrix.ocaml-version }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-version }}
+          dune-cache: ${{ matrix.os != 'macos-latest' }}
+
+      - name: Install dependencies
+        run: opam install . --deps-only --with-doc --verbose
```
Comment (Nymphium) [inline .github/workflows/ci.yml]: This repo assumes to be built on Nix, so when CI uses opam it needs `opam pin add ocaml-protoc-plugin https://github.com/Nymphium/ocaml-protoc-plugin#create-rpc-path`.
Additionally, it requires `grpcurl`, `protobuf` and `yarn` for tests.
```suggestion
        run: |
          apt install protobuf grpcurl yarn # I don't know ubuntu has these pkgs or its correct name
          opam pin add ocaml-protoc-plugin https://github.com/Nymphium/ocaml-protoc-plugin#create-rpc-path
          opam install . --deps-only --with-doc --verbose
```

**Review by Nymphium — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Install dependencies
+        run: |
+          if [ "${{ matrix.os }}" = "ubuntu-latest" ] ; then
+            sudo apt-get install -y protobuf grpcurl yarn
+          else
+            brew install protobuf grpcurl yarn
+          fi
+          
+          opam pin add ocaml-protoc-plugin https://github.com/Nymphium/ocaml-protoc-plugin#create-rpc-path
```
Comment (Nymphium) [inline .github/workflows/ci.yml]: It must has `.git`, my mistake
```suggestion
          opam pin add ocaml-protoc-plugin https://github.com/Nymphium/ocaml-protoc-plugin.git#create-rpc-path
```

File: grpc.opam [hunk 3]
```diff
@@ -17,6 +17,7 @@ depends: [
   "ppx_deriving"
   "ppx_yojson_conv"
   "ppx_let"
+  "fmt"
```
Comment (Nymphium) [inline grpc.opam]:
```suggestion
  "fmt" { with-test }
```

**Comment by y-yu** [comment]
Success 🎉 
- https://github.com/y-yu/grpc-ocaml/runs/6797944105?check_suite_focus=true

File: grpc.opam (hunk 3 above)
Comment (y-yu) [reply grpc.opam]: 🤔 

```
Run opam exec -- dune build
File "example/dune", line 3, characters 45-52:
3 |  (libraries grpc.server grpc.client logs.fmt fmt.tty))
                                                 ^^^^^^^
Error: Library "fmt.tty" not found.
-> required by _build/default/example/client.exe
-> required by alias example/all
-> required by alias default
```

https://github.com/y-yu/grpc-ocaml/runs/6797798509?check_suite_focus=true#step:5:1

Comment (Nymphium) [reply grpc.opam]: Sorry I myself misunderstood that `fmt` is not for test but for example😜

```suggestion
```

In addition, Dune automatically detects the build dependencies, so we dont't need `dune build` before `dune runtest`.

Comment (y-yu) [reply grpc.opam]: OK, I remove `dune build` and `fmt`.

**Review by Nymphium — APPROVED** [review]
Thank you🤗

**Pull request merged**